TPP.ArgsParsing/TypeParsers/RoleParser.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TPP.Persistence.Models;
+
+namespace TPP.ArgsParsing.TypeParsers
+{
+    /// <summary>
+    /// A parser that finds a role by name.
+    /// </summary>
+    public class RoleParser : BaseArgumentParser<Role>
+    {
+        public override Task<ArgsParseResult<Role>> Parse(IImmutableList<string> args, Type[] genericTypes)
+        {
+            string roleToParse = args[0];
+            ArgsParseResult<Role> result;
+            try
+            {
+                Role parsedRole = (Role)Enum.Parse(typeof(Role), roleToParse, true);
```

Comment:
Small QOL improvement so people don't have to check the documentation:
```suggestion
                Role parsedRole = (Role)Enum.Parse(typeof(Role), roleToParse, ignoreCase: true);
```